### PR TITLE
daemonset pkg: WithPodAffinity method added

### PR DESCRIPTION
**Pull request by elenagerman**

### CODE REVIEW

File: pkg/daemonset/daemonset.go [hunk 1]
```diff
@@ -174,6 +174,20 @@ func (builder *Builder) WithHostNetwork() *Builder {
...
 
+// WithPodAffinity applies pod's Affinity to daemonset definition.
+func (builder *Builder) WithPodAffinity(podAffinity *corev1.Affinity) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Definding pod affinity to daemonset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
```
Comment (klaskosk) [inline pkg/daemonset/daemonset.go:185]: could you add a check that podAffinity isn't nil please? (and update the test)

**Comment by elenagerman** [comment]
@kononovn @klaskosk @trewest @cdvultur @achuzhoy PTAL

File: pkg/daemonset/daemonset.go [hunk 2]
```diff
@@ -174,6 +174,28 @@ func (builder *Builder) WithHostNetwork() *Builder {
...
+// WithPodAffinity applies pod's Affinity to daemonset definition.
+func (builder *Builder) WithPodAffinity(podAffinity *corev1.Affinity) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if podAffinity == nil {
+		glog.V(100).Infof("The Affinity parameter is empty")
+
+		builder.errorMsg = "Affinity parameter is empty"
```
Comment (klaskosk) [inline pkg/daemonset/daemonset.go:186]: one last minor point but errorMsg should start with a lowercase
```suggestion
		builder.errorMsg = "affinity parameter is empty"
```

**Review by klaskosk — COMMENTED** [review]
one last minor comment but otherwise looks good, thanks for adding the validation

**Comment by elenagerman** [comment]
@kononovn @trewest @cdvultur @achuzhoy PTAL

**Review by trewest — APPROVED** [review]
/lgtm

**Pull request merged**